app/reducers/activity.js

```diff
@@ -266,91 +255,111 @@ function groupAll(data) {
+      (item.tx_hash && item.tx_hash.includes(searchText)) ||
+      (item.payment_hash && item.payment_hash.includes(searchText)) ||
+      (item.payment_request && item.payment_request.includes(searchText))
+  )
+}
+
 const allActivity = createSelector(
   searchSelector,
   paymentsSending,
   transactionsSending,
   paymentsSelector,
   transactionsSelector,
   invoicesSelector,
-  showExpiredSelector,
-  (
-    searchText,
-    paymentsSending,
-    transactionsSending,
-    payments,
-    transactions,
-    invoices,
-    showExpired
-  ) => {
-    const filteredInvoices = invoices.filter(
-      invoice => showExpired || invoice.settled || !invoiceExpired(invoice)
-    )
-
+  (searchText, paymentsSending, transactionsSending, payments, transactions, invoices) => {
     const allData = [
       ...paymentsSending,
       ...transactionsSending,
       ...payments,
       ...transactions.filter(transaction => !transaction.isFunding && !transaction.isClosing),
-      ...filteredInvoices
+      ...invoices.filter(invoice => invoice.settled || !invoiceExpired(invoice))
     ]
-
-    if (!searchText) {
-      return groupAll(allData)
-    }
-
-    const searchedArr = allData.filter(
-      tx =>
-        (tx.tx_hash && tx.tx_hash.includes(searchText)) ||
-        (tx.payment_hash && tx.payment_hash.includes(searchText)) ||
-        (tx.payment_request && tx.payment_request.includes(searchText))
-    )
-
-    return groupAll(searchedArr)
+    return searchText ? groupAll(applySearch(allData, searchText)) : groupAll(allData)
   }
 )
 
-const invoiceActivity = createSelector(
-  invoicesSelector,
-  showExpiredSelector,
-  (invoices, showExpired) =>
-    groupAll(invoices.filter(invoice => showExpired || invoice.settled || !invoiceExpired(invoice)))
-)
-
 const sentActivity = createSelector(
+  searchSelector,
   paymentsSending,
   transactionsSending,
   paymentsSelector,
   transactionsSelector,
-  (paymentsSending, transactionsSending, payments, transactions) => {
-    return groupAll([
+  (searchText, paymentsSending, transactionsSending, payments, transactions) => {
+    const allData = [
       ...paymentsSending,
       ...transactionsSending,
       ...payments,
-      ...transactions.filter(transaction => !transaction.received)
-    ])
+      ...transactions.filter(
+        transaction => !transaction.received && !transaction.isFunding && !transaction.isClosing
+      )
+    ]
+    return searchText ? groupAll(applySearch(allData, searchText)) : groupAll(allData)
+  }
+)
+
+const receivedActivity = createSelector(
+  searchSelector,
+  invoicesSelector,
+  transactionsSelector,
+  (searchText, invoices, transactions) => {
+    const allData = [
+      ...invoices.filter(invoice => invoice.settled),
+      ...transactions.filter(
+        transaction => transaction.received && !transaction.isFunding && !transaction.isClosing
+      )
+    ]
+    return searchText ? groupAll(applySearch(allData, searchText)) : groupAll(allData)
```

Comment:
I'm seeing this routine repeating multiple times in this module. Might be a good idea to extract it into a helper function

Reply:
Updated